### DISCUSSION TITLE
Issue upgrading from 0.3.1 to 0.4.x

### DISCUSSION THREAD

**Discussion by Neighborkid01** [leptos-rs/leptos#1897] [Q&A] [answered]
Hello,

I have a project I haven't been able to work on in a while and am just getting back to it. I'm trying to upgrade from Leptos 0.3.1 to a later version (eventually 0.5.0).
I wanted to move to 0.4.x and fix any issues and then move from there to 0.5.
My current `Cargo.toml` and the version I'm migrating from are [here](https://github.com/Neighborkid01/Leptos-upgrade-question)

When I run `cargo leptos build`, I get the following error:
```
error[E0599]: no method named `line` found for struct `proc_macro::LineColumn` in the current scope
   --> /Users/razor_crest/.cargo/registry/src/index.crates.io-6f17d22bba15001f/leptos_macro-0.4.10/src/lib.rs:397:30
    |
397 |                 site.start().line()
    |                              ^^^^-- help: remove the arguments
    |                              |
    |                              field, not a method

For more information about this error, try `rustc --explain E0599`.
```

I've tried adding `leptos_macro` to my `Cargo.toml` so that I can explicitly set the version, but I still get the same error on other 0.4.x versions.

If I only update the Leptos version to 0.4 and don't update it to have `features = ["nightly"]`, I get syntax errors for my components as expected, but as soon as I add the nightly feature to `leptos`, `leptos_meta`, or `leptos_router`, I get the error.

Any help would be appreciated as I'm not quite sure what dependency this issue actually stems from. I don't know if it's from `leptos_macro` or somehow in `proc_macro`, or if I need to pin a specific version of a dependency...

Thanks!

**Comment by gbj** [comment, marked as answer]
You should just need to `rustup update nightly`. There was a breaking change in this nightly API in about late July, which caused this issue until people updated Leptos version and nightly compiler.

**Comment by Neighborkid01** [comment]
Thank you!
